cmd/mount.go

```diff
@@ -106,7 +106,32 @@ func (opt *mountOptions) validate(skipNonEmptyMount bool) error {
 	if _, err := os.Stat(opt.MountPath); os.IsNotExist(err) {
 		return fmt.Errorf("mount directory does not exists")
 	} else if common.IsDirectoryMounted(opt.MountPath) {
-		return fmt.Errorf("directory is already mounted")
+		// Try to cleanup the stale mount
+		log.Info("Mount::validate : Mount directory is already mounted, trying to cleanup")
+		active, err := common.IsMountActive(opt.MountPath)
+		if active || err != nil {
+			// Previous mount is still active so we need to fail this mount
+			return fmt.Errorf("directory is already mounted")
+		} else {
+			// Previous mount is in stale state so lets cleanup the state
+			log.Info("Mount::validate : Cleaning up stale mount")
+			if err = unmountBlobfuse2(opt.MountPath); err != nil {
+				return fmt.Errorf("directory is already mounted, unmount manually before remount.")
```

Comment:
```suggestion
				return fmt.Errorf("directory is already mounted, unmount manually before remount [%s]", err.Error())
```

Reply:
For any reason if we failed to unmount, we do not want user to have more info on the console and rather ask them to try to unmount manually. Hence I skipped adding the err.Error() here intentionally. If user gets any sort of error like permission denided or anything, it will not make any sense to it.